connect/transforms/src/test/java/org/apache/kafka/connect/transforms/ExtractFieldTest.java

```diff
@@ -186,7 +189,98 @@ public void nonExistentNestedFieldWithSchemaShouldFail() {
 
     @Test
     public void testExtractFieldVersionRetrievedFromAppInfoParser() {
-        assertEquals(AppInfoParser.getVersion(), xform.version());
+        assertEquals(AppInfoParser.getVersion(), xformKey.version());
+    }
+
+    @Test
+    @DisplayName("Null key value instead of default must be used")
+    public void whenKeyConverterReplaceNullWithDefaultIsFalseExtractFieldWithNullValueMustReturnNull() {
```

Comment:
Can we simplify the test name? What about `testUnsetOptionalFieldIsNull()`?
Same in the other file below.